### PR TITLE
User fewer Vector internals

### PR DESCRIPTION
**Pull request by danieldk**
## Description

Throughout spaCy, we rely on internals of the `Vector` class. This is a first PR to replace a bunch of such uses by methods/properties. In particular:

* `Vectors.data.shape` -> `Vectors.shape`
* `Vectors.data.size` -> `Vectors.size`
*  `self.vocab.vectors.data = ops.asarray(self.vocab.vectors.data)` -> `self.to_ops(ops)`

This was the lower-hanging fruit 🥝, I'll make a separate PR for more difficult cases.

### Types of change

Cleanup.

## Checklist

- [x] I confirm that I have the right to submit this contribution under the project's MIT license.
- [x] I ran the tests, and all new and existing tests passed.
- [x] My changes don't require a change to the documentation, or if they do, I've added all required information.

### CODE REVIEW

**Review by svlandeg — APPROVED** [review]
I love this kind of PR 🤩 
All looks perfect!

Can't wait for the rest 🍎 🍌 🍇

File: spacy/vectors.pyx [hunk 1]
```diff
@@ -511,6 +511,9 @@ cdef class Vectors:
                     for i in range(len(queries)) ], dtype="uint64")
         return (keys, best_rows, scores)
 
+    def to_ops(self, ops: Ops):
```
Comment (svlandeg) [inline spacy/vectors.pyx:514]: Perhaps we can document this one at https://spacy.io/api/vectors ?

Comment (svlandeg) [reply spacy/vectors.pyx:514]: I'm not sure how often it would come up that a user would need to call this directly though.

Comment (danieldk) [reply spacy/vectors.pyx:514]: I don't know. But I think it doesn't hurt!

**Comment by danieldk** [comment]
@explosion-bot please test_gpu

**Comment by adrianeboyd** [comment]
I think it would make sense to add the new `to_ops` to the deserialization here, too, which will be needed for the bigendian ops, see: https://github.com/explosion/murmurhash/issues/26#issuecomment-976565538

**Comment by danieldk** [comment]
> I think it would make sense to add the new `to_ops` to the deserialization here, too, which will be needed for the bigendian ops, see: [explosion/murmurhash#26 (comment)](https://github.com/explosion/murmurhash/issues/26#issuecomment-976565538)

I wonder if it wouldn't be better to provide an `Ops.load` which calls `xp.load` and then does an endianness check + conversion. Then all code that loads data would be robust against different endianness.

**Pull request merged**